### ISSUE TITLE
Add support for WooCommerce productCategories

### ISSUE THREAD

**Issue by alexkey89** [AxeWP/wp-graphql-rank-math#40]
### What problem does this address?

It doesn't seem to return SEO for woocommerce products whether if is simple product or variable. 

### What is your proposed solution?

Add support / extend this plugin to support WooCommerce better

### What alternatives have you considered?

YoastSEO already provides graphql support.

**Comment by justlevine** [comment]
@alexkey89 just to be clear, are you referring to supporting vanilla WooCommerce, or are you also using WooGraphQL to expose the types?

**Comment by alexkey89** [comment]
Well, with yoastSEO it was returning in variable products and simple the opengraph etc etc. Im also using woographql but doesn't seem to have support for rankmath out of the box. e.g in yoast SEO i can do that: 

`export const PRODUCT_BY_SLUG_QUERY = gql
  query Product($slug: ID!) {
    product(id: $slug, idType: SLUG) {
      productId: databaseId
      status
      type
      averageRating
      slug
      sku
      priceHtml
      productCategories {
        nodes {
          parent {
            node {
              name
              slug
              parentDatabaseId
            }
          }
          name
          slug
        }
      }
      description
      shortDescription
      galleryImages {
        nodes {
          altText
          mediaItemUrl
        }
      }
      attributes {
        nodes {
          name
          options
        }
      }
      image {
        altText
        sourceUrl
      }
      related {
        nodes {
          ... on VariableProduct {
            regularPrice
            salePrice
          }
          ... on SimpleProduct {
            regularPrice
            salePrice
          }
          slug
          productId: databaseId
          name
          image {
            sourceUrl
            altText
          }
          allPaBrand {
            nodes {
              name
              slug
            }
          }
        }
      }
      name
      ... on SimpleProduct {
        price
        id
        regularPrice
        stockStatus
        stockQuantity
        salePrice
      }
      ... on VariableProduct {
        price
        id
        regularPrice
        salePrice
seo {
  title
  canonical
  metaDesc
  opengraphUrl
  opengraphTitle
  opengraphDescription
  opengraphImage {
    altText
    guid
    mediaDetails {
      sizes {
        height
        width
        name
        sourceUrl
      }
    }
  }
  opengraphSiteName
  twitterTitle
  twitterDescription
  twitterImage {
    altText
    mediaType
    uri
    title(format: RENDERED)
  }
}
        variations(where: { stockStatus: IN_STOCK }) {
          nodes {
            name
            variationId: databaseId
            stockStatus
            stockQuantity
            price
            regularPrice
            salePrice
            attributes {
              nodes {
                value
              }
            }
          }
        }
      }
      ... on ExternalProduct {
        price
        id
        regularPrice
        salePrice
        externalUrl
      }
      ... on GroupProduct {
        products {
          nodes {
            ... on SimpleProduct {
              id
              price
              regularPrice
              salePrice
            }
          }
        }
      }
    }
  }
`;`

I could add this inside variant product and get the desired SEO data back: 
`seo {
  title
  canonical
  metaDesc
  opengraphUrl
  opengraphTitle
  opengraphDescription
  opengraphImage {
    altText
    guid
    mediaDetails {
      sizes {
        height
        width
        name
        sourceUrl
      }
    }
  }
  opengraphSiteName
  twitterTitle
  twitterDescription
  twitterImage {
    altText
    mediaType
    uri
    title(format: RENDERED)
  }
}`

but in rankmath I do this: 
`          ... on VariableProduct {
            productId: databaseId
            price
            onSale
            regularPrice
            salePrice
             seo{
              title
            }` and get internal server error due to the fact that possibly they are not registered? 

screenshot below
![Screenshot 2023-03-15 at 10 58 44 (2)](https://user-images.githubusercontent.com/5107980/225260569-cd2e432d-9e2e-4c34-a16f-f28638d233a8.png)

**Comment by alexkey89** [comment]
@justlevine can support be added any time soon?

**Comment by justlevine** [comment]
@alexkey89 I just spun up a new instance with WPGraphQL Woocommerce v0.12.3 (used woo's dummy data csv and then added a variable product), and everything seems to be working as expected.

![image](https://user-images.githubusercontent.com/29322304/230468759-9088f017-2395-48e5-9aa3-73cf04b6a155.png)

If you're getting an internal server error, it sounds like you're running into a bug with something else, but I don't have nearly enough information to replicate. Can you start by sharing what the internal server error actually says? It'l be in your `debug.log`.

**Comment by alexkey89** [comment]
Thanks! Took me a while, but It seems issue lies with wpgraphql SEO add-on. It interferes with this one and causes internal server error. As the name suggests WPGraphQL SEO, I didn't predict that that plugin was intended for yoastSEO only. Either way, I suggest to have a validation to block activation of that plugin if WPGraphQL Rank Math plugin is already installed and vice versa.

**Comment by alexkey89** [comment]
Also sorry to comment again, but it seems that when it comes to productcategories, it doesn't give the title the url properly: 
```
query FILTER_ALL {
  productCategory(id: "tops-t-shirts", idType: SLUG) {
    name
    description
    databaseId
    slug
    seo {
      title
      breadcrumbTitle
      canonicalUrl
      description
      focusKeywords
      openGraph {
        url
        siteName
        twitterMeta {
          title
          site
          image
          card
          description
        }
        title
        description
        image {
          secureUrl
          height
          type
          width
        }
      }
    }
    parent {
      node {
        name
        slug
        parentDatabaseId
      }
    }
    children(first: 100) {
      edges {
        node {
          name
          slug
        }
      }
    }
    categoryacfs {
      promoCategories {
        name
        slug
      }
    }
  }
  products(
    first: 5
    after: null
    last: null
    before: null
    where: {minPrice: 1, maxPrice: 300, status: "publish", stockStatus: IN_STOCK, orderby: {field: PRICE, order: DESC}}
  ) {
    pageInfo {
      total
      hasNextPage
      hasPreviousPage
      endCursor
      startCursor
    }
    edges {
      node {
        date
        name
        slug
        discountTableAndBar
        status
        priceHtml
        productId: databaseId
        attributes {
          nodes {
            name
            options
          }
        }
        image {
          sourceUrl(size: LARGE)
          altText
        }
        productCategories {
          edges {
            node {
              categoryacfs {
                promoCategories {
                  name
                  slug
                }
              }
              name
              slug
              count
              name
              description
              children {
                nodes {
                  name
                  slug
                  name
                  description
                  count
                }
              }
            }
          }
        }
        name
        slug
        status
        date
        ... on VariableProduct {
          productId: databaseId
          price
          onSale
          regularPrice
          salePrice
          variations(where: {stockStatus: IN_STOCK}) {
            edges {
              node {
                attributes {
                  nodes {
                    value
                  }
                }
              }
            }
          }
        }
        ... on SimpleProduct {
          productId: databaseId
          price
          onSale
          regularPrice
          salePrice
        }
        ... on GroupProduct {
          products {
            nodes {
              ... on SimpleProduct {
                id: databaseId
                price
                onSale
                regularPrice
                salePrice
              }
            }
          }
        }
        ... on ExternalProduct {
          productId: databaseId
          price
          onSale
          regularPrice
          salePrice
        }
        allPaSize {
          nodes {
            name
            databaseId
            slug
          }
        }
        allPaBrand {
          nodes {
            name
            databaseId
            slug
          }
        }
        allPaFit {
          nodes {
            name
            databaseId
            slug
          }
        }
        allPaMaterial {
          nodes {
            name
            databaseId
            slug
          }
        }
        allPaProductShape {
          nodes {
            name
            databaseId
            slug
          }
        }
        allPaColor {
          nodes {
            name
            databaseId
            slug
          }
        }
      }
    }
  }
}
```
That's the request but the response comes like this: (I'm using latest wpgraphl and wpgraphl woocommerce as of 18/4/2023)
```
{
  "data": {
    "productCategory": {
      "name": "Tops & T-shirts",
      "description": "<span data-sheets-value=\"{\" data-sheets-userformat=\"{\">Discover our everyday collection of women’s tops, from classics to tees and tanks. Choose cotton fabrics for your comfort or soft fabrics for warmer days.</span>",
      "databaseId": 2972,
      "slug": "tops-t-shirts",
      "seo": {
        "title": null,
        "breadcrumbTitle": "Tops &amp; T-shirts",
        "canonicalUrl": null,
        "description": null,
        "focusKeywords": null,
        "openGraph": {
          "url": null,
          "siteName": " | Online Clothing Store",
          "twitterMeta": {
            "title": null,
            "site": null,
            "image": "my-site/logo.png",
            "card": "SUMMARY_LARGE_IMAGE",
            "description": null
          },
          "title": null,
          "description": null,
          "image": {
            "secureUrl": "my-site/logo.png",
            "height": 1000,
            "type": "image/png",
            "width": 1000
          }
        }
      },
      "parent": {
        "node": {
          "name": "Womens",
          "slug": "womens",
          "parentDatabaseId": null
        }
      },
      "children": {
        "edges": []
      },
      "categoryacfs": {
        "promoCategories": null
      }
    },
    "products": {
      "pageInfo": {
        "total": 12288,
        "hasNextPage": true,
        "hasPreviousPage": false,
        "endCursor": "YXJyYXljb25uZWN0aW9uOjc5NzE4Mg==",
        "startCursor": "YXJyYXljb25uZWN0aW9uOjc4NzY3Nw=="
      },......
```
When it comes the individual product page, it shows correctly, issue is with product categories

**Comment by justlevine** [comment]
Fixed by #57 

![image](https://github.com/AxeWP/wp-graphql-rank-math/assets/29322304/34d85e99-8505-4596-8e0e-b233f431ec3f)

**Comment by justlevine** [comment]
https://github.com/AxeWP/wp-graphql-rank-math/releases/tag/0.0.14
